### PR TITLE
Request.legacy_arguments & LegacyMultiDict: collapsing arguments fix

### PR DESCRIPTION
**Pull request by popravich**

### CODE REVIEW

File: zorro/web.py [hunk 1]
```diff
@@ -55,18 +61,22 @@ def parsed_uri(self):
...
         return arguments
 
     @cached_property
+    def form_arguments(self):
+        return dict(self._argument_tuples)
+
+    @cached_property
     def legacy_arguments(self):
-        return LegacyMultiDict(self.form_arguments)
+        return LegacyMultiDict(self._argument_tuples)
```
Comment (tailhook) [inline zorro/web.py]: Just feed result of parse_qsl here, don't need another list as a proxy

File: zorro/web.py [hunk 2]
```diff
@@ -40,6 +43,9 @@ def __iter__(self):
         for k in self._dic:
             yield k
 
+    def __getitem__(self, k):
```
Comment (tailhook) [inline zorro/web.py]: Does it resemble right semantics of **getitem** of multi-dict?

File: zorro/web.py [hunk 3]
```diff
@@ -27,11 +27,14 @@ class LegacyMultiDict(object):
     """Utilitary class which wrap dict to make it suitable for old utilities
     like wtforms"""
 
-    def __init__(self, dic):
-        self._dic = dic
+    def __init__(self, pairs):
+        dic = defaultdict(list)
+        for k, v in pairs:
+            dic[k].append(v)
+        self._dic = dict(dic)
```
Comment (tailhook) [inline zorro/web.py]: defaultdict is handy but double-copying it is bad, white original loop on normal dict, it's not that hard

File: zorro/web.py [hunk 4]
```diff
@@ -27,11 +27,14 @@ class LegacyMultiDict(object):
...
 
     def getlist(self, k):
-        return [self._dic[k]]
+        return list(self._dic[k])
```
Comment (tailhook) [inline zorro/web.py:42]: Do you really need a copy here?

Comment (popravich) [inline zorro/web.py:42]: as seen in Werkzeug and WebOb (just a foolproof)

**Comment by popravich** [comment]
Updated.
- removed `defaultdict`
- removed `LegacyMultiDict.__getitem__`
- added `LegacyMultiDict.__len__`
- added `LegacyMultiDict.update`

**Pull request merged**